### ISSUE TITLE
If `maxsize`>`nobs` should not return an error, should just do a single batch

### ISSUE THREAD

**Issue by oxinabox** [JuliaML/MLDataPattern.jl#11]
Example:

```julia
julia> typeof(ds)
Array{Int64,2}

julia> nobs(ds)
1_523_108                        # manually separated for readability

julia> eachbatch(ds, maxsize=5_000_000) |> first
ERROR: ArgumentError: Specified batch-size is too large for the given number of observations
 in _compute_batch_settings(::Array{Int64,2}, ::Int64, ::Int64, ::LearnBase.ObsDim.Last, ::Bool) at /home/ubuntu/.julia/v0.5/MLDataPattern/src/dataview.jl:172
 in MLDataPattern.BatchView{TElem,TData,O}(::Array{Int64,2}, ::Int64, ::Int64, ::LearnBase.ObsDim.Last, ::Bool) at /home/ubuntu/.julia/v0.5/MLDataPattern/src/dataview.jl:354
 in #eachbatch#118(::Int64, ::Int64, ::Int64, ::LearnBase.ObsDim.Last, ::Function, ::Array{Int64,2}) at /home/ubuntu/.julia/v0.5/MLDataPattern/src/dataiterator.jl:550
 in (::MLDataPattern.#kw##eachbatch)(::Array{Any,1}, ::MLDataPattern.#eachbatch, ::Array{Int64,2}) at ./<missing>:0
```

What has been said is "Break this into evenly sized Batches of length no more than 5_000_000 observations long."

It has errorred  saying that 5_000_000 is more than the number of observations.
But that doesn't matter, because it is not an issue for max_size.

There is indeed a batch size that is less than 5_000_000, that evenly divides the observations:
--- 1_523_108.
Taking it all in a single Batch.

I believe this would be more intuitive behaviour.

**Comment by Evizero** [comment]
mhm, I agree. Should be an easy fix

**Comment by Evizero** [comment]
fixed in #12
